### PR TITLE
[ENH] Move row column

### LINKED ISSUE #387: [ENH] Send row or column to an arbitrary position in the sequence of rows/columns
# Brief Description

I would like to propose a new function that moves a row (or column) that a user picks to a new position in the sequence of rows/columns.

cc: @bhallaY: Can you fill out an example API?

### PR DESCRIPTION
**Pull request by asearfos**
# PR Description

Please describe the changes proposed in the pull request: 

- added function which moves a row or column to new position in the dataframe
- this does not support multi-level indexing

**This PR resolves #387.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:master, but rather from `<your_username>`:<branch_name>.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

## Quick Check

To do a very quick check that everything is correct, follow these steps below:

- [x] Run the command `make check` from pyjanitor's top-level directory. This will automatically run:
    - black formatting
    - pycodestyle checking
    - running the test suite
    - docs build
    
Once done, please check off the check-box above.
    
If `make check` does not work for you, you can execute the commands listed in the Makefile individually.

## Code Changes

If you are adding code changes, please ensure the following:

- [ ] Ensure that you have added tests.
- [ ] Run all tests (`$ pytest .`) locally on your machine.
    - [ ] Check to ensure that test coverage covers the lines of code that you have added.
    - [ ] Ensure that all tests pass.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -25,6 +25,71 @@
...
+    position: str = "before",
+    axis: int = 0,
+) -> pd.DataFrame:
+    """
+     Move column or row to a position adjacent to another column or row in dataframe.
+
+     :param df: The pandas Dataframe object.
+     :param int or str source: column or row to move
+     :param str target: column or row to move adjacent to
+     :param str position: Specifies whether the series is moved to before or after the adjacent series. Values can be either 'before' or 'after'; defaults to 'before'.
```
Comment (ericmjl) [inline janitor/functions.py]: The lines that are greater than 79 characters are what the CI build is catching as errors. Could you reformat docstring lines that are >79 chars? One example is this one.

```suggestion
     :param str position: Specifies whether the series is moved to before or 
        after the adjacent series. Values can be either 'before' or 'after'; 
        defaults to 'before'.
```

I will mark the other lines that are causing errors.

File: janitor/functions.py [hunk 2]
```diff
@@ -25,6 +25,71 @@
...
+@pf.register_dataframe_method
+def move(
+    df: pd.DataFrame,
+    source: Union[int, str],
+    target: Union[int, str],
+    position: str = "before",
+    axis: int = 0,
+) -> pd.DataFrame:
+    """
+     Move column or row to a position adjacent to another column or row in dataframe.
```
Comment (ericmjl) [inline janitor/functions.py]: This one is also erroring.

File: janitor/functions.py [hunk 3]
```diff
@@ -25,6 +25,71 @@
...
+
+     :param df: The pandas Dataframe object.
+     :param int or str source: column or row to move
+     :param str target: column or row to move adjacent to
+     :param str position: Specifies whether the series is moved to before or after the adjacent series. Values can be either 'before' or 'after'; defaults to 'before'.
+     :param int axis: Axis along which the function is applied. 0 to move a row, 1 to move a column.
```
Comment (ericmjl) [inline janitor/functions.py]: This one is also erroring.

File: janitor/functions.py (hunk 2 above)
Comment (asearfos) [reply janitor/functions.py]: corrected

**Pull request merged**